proto/structured.proto

```diff
@@ -104,3 +105,220 @@ message CreateTableResponse {
+// - If key doesn't exist and exp_row is empty, sets row.
+// - If key exists and exp_row is empty, sets row.
+// - Otherwise, returns error and the actual value of the row in the response.
+message ConditionalPutTableRequest {
+  optional RequestHeader header = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
+  optional TableKey key = 2 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
+  // The values to put.
+  optional TableRow row = 3 [(gogoproto.nullable) = false];
+  optional TableRow exp_row = 4[(gogoproto.nullable) = false];
+}
+
+// A ConditionalPutTableResponse is the return value from the
+// ConditionalPut request.
+message ConditionalPutTableResponse {
+  optional TableResponseHeader header = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
+  optional TableRow row = 2[(gogoproto.nullable) = false];
+}
+
+// A DeleteRowTableRequest.
+message DeleteRowTableRequest {
+  optional TableRequestHeader header = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
+  optional TableKey key = 2 [(gogoproto.nullable) = false];
+}
+
+// A DeleteRowTableResponse.
+message DeleteRowTableResponse {
+  optional TableResponseHeader header = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
+}
+
+// A DeleteRowRangeTableRequest deletes the specified rows.
+message DeleteRowRangeTableRequest {
+  optional TableRequestHeader header = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
+  // Delete *all* entries between Key (inclusive) and EndKey
+  // (exclusive). Must be > 0
+  optional string index = 2 [(gogoproto.nullable) = false];
+  optional TableKey key = 3 [(gogoproto.nullable) = false];
+  optional TableKey end_key = 4 [(gogoproto.nullable) = false];
+  optional int64 max_entries_to_delete = 5 [(gogoproto.nullable) = false];
+}
+
+// A DeleteRowRangeTableResponse is the response.
+message DeleteRowRangeTableResponse {
+  optional TableResponseHeader header = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
+  // Number of entries deleted.
+  optional int64 num_deleted = 2 [(gogoproto.nullable) = false];
+}
+
+// A ScanTableRequest. It specifies the start and end keys for the scan
+// and the maximum number of results.
+message ScanTableRequest {
+  optional TableRequestHeader header = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
+  // Must be > 0.
+  optional string index = 2 [(gogoproto.nullable) = false];
+  optional TableKey key = 3 [(gogoproto.nullable) = false];
+  optional TableKey end_key = 4 [(gogoproto.nullable) = false];
+  optional int64 max_results = 5 [(gogoproto.nullable) = false];
+}
+
+// A ScanTableResponse is the response.
+message ScanTableResponse {
+  optional TableResponseHeader header = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
+  // Empty in case of error.
+  repeated TableRow rows = 2;
+}
+
+// A TableRequestUnion contains exactly one of the optional requests.
+message TableRequestUnion {
```

Comment:
seems like you'd want these to be nested inside the Batch{Request,Response}, but i dont feel strongly